### ISSUE TITLE
An options to have more themes, for the visual look, of the editor . .

### ISSUE THREAD

**Issue by jasperbrooks79** [godotengine/godot-proposals#1417]
**Describe the project you are working on:**
A Tomb Raider game, atm . .

**Describe the problem or limitation you are having in your project:**
I really like the current look, of the engine, it's very smart, I was wondering if we could have themes, where all the docs have rounded corners, so they look, a bit, like bubbles . . The old program, WinAmp, a mp3 player, had a bunch of cool themes, it could be cool, if there were a few nice ones, with each release of Godot, we get another one, or two, for mile-stone releases, so one can make it look, even better . . Lot of work, but maybe the new user interface expert, could make something nice, over time . . Lots, of work, finding a new one, for each major release, will be a challenge, look to WinAmp, for ideas . . .

![2020-08-26 1813](https://user-images.githubusercontent.com/61069740/91329091-09ac8c80-e7c8-11ea-9c55-87ab9a696e42.gif)

![2020-08-26 1814](https://user-images.githubusercontent.com/61069740/91329110-0dd8aa00-e7c8-11ea-8d98-ecb896658ce6.gif)

![2020-08-26 1811](https://user-images.githubusercontent.com/61069740/91328762-9dca2400-e7c7-11ea-882c-3a615f16a3ab.png)

Finding a really cool theme, for each new release, could be really amazing, so we always have a surprise, a cool one, for new releases, imagine what Godot could look like, with in-built skins, like that . .

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Just, for fun, maybe the Godot engine team could have a few people, work on some ideas, then before launch, pick the best ones, or so . . Really made WinAmp more fun and, epic . .

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Not sure, like a skin, or theme, perhaps drawing inspiration from new features, could be neat . .

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Not sure, not engine expert . .

**Is there a reason why this should be core and not an add-on in the asset library?:**
Well, it'd mostly be for looks and, effect . . But, it could be a chance, for Godot team, to make something a bit more ' wild ', to mark each new release, or like, so . .

:D <3 <3 . .

**Comment by RaTcHeT302** [comment]
i'll ignore the ratings since, you guys aren't actually making any points or anything, ur just lazily downvoting an actually good idea, i'm not going to take part into the bandwagon
![lazy](https://user-images.githubusercontent.com/43797325/91420237-116a4080-e85d-11ea-8aeb-29c1f6695d8e.png)

i don't think such themes will ever be included by default, but i would totally encourage people to make total conversion themes, but does the current user interface in Godot allow for such radical changes? i think it would be very nice if we could customize the user interface to this extent

![PqXdoRL](https://user-images.githubusercontent.com/43797325/91420197-fc8dad00-e85c-11ea-930d-b2faf690e8b6.png)

i would love to have some kind of easy "drag and drop" system, where you can just move things around, and customize the look of everything as some sort of global property (the colors, the fonts, if the panel should have a background image, etc)

i think allowing users to go wild with the user interface would be really nice, and i think the core idea is a good one

i think having an easy, visual editor, for the user interface, which allows you to go batshit crazy would be really great

i don't know how the user interface in WinAmp works, i don't think we should have skins or themes included by default, since some of them are kinda gaudy \ ugly, but i do approve of the idea, of being able to customize LITERALLY EVERYTHING very easily

i don't have the time to come up with a concept, but it would be nice to have a user interface editor or an user interface designer system
![example](https://user-images.githubusercontent.com/43797325/91420781-cd2b7000-e85d-11ea-906d-6e420a6c76b4.png)

tl:dr: yes, allowing the user interface, to be fully customizable, would be a very good thing, we just need a better system for it

**Comment by Xrayez** [comment]
I honestly think that Godot could benefit from having more fine-looking editor themes. The new default scheme created in 3.0 made Godot quite cold compared to 2.1...

I propose we should have a proper dark theme for the editor at least. See examples in https://github.com/Xrayez/godot-editor-settings.

**Comment by TheDuriel** [comment]
This dark theme already exists. I've been using it for years now, and keep getting compliments and shoutouts for it.

Go into the Editor Settings. Theme, choose Dark. Set borders to 0, funfact you can make rounded corners here too. And the color to whatever you want. I use pure red.

**Comment by Calinou** [comment]
> funfact you can make rounded corners here too.

Not yet (unless you create a custom editor theme from scratch). This remains to be implemented.

**Comment by umarcor** [comment]
@RaTcHeT302,

> i'll ignore the ratings since, you guys aren't actually making any points or anything, ur just lazily downvoting an actually good idea, i'm not going to take part into the bandwagon

I guess that most are genuinely confused because theming and templating is a built-in feature. In fact, the only purpose of Godot is to build videogames, which are essentially custom templates for interfaces, as opposed to other GUI toolkits which provide less customization features. Moreover, the editor and assets are written with the engine itself. Many assets are, indeed, custom templates for doing specific tasks in the editor.

Hence, I guess we need something more explicit than WinAmp screenshots (which otherwise brings back very nice memories).

> i don't think such themes will ever be included by default, but i would totally encourage people to make total conversion themes, but does the current user interface in Godot allow for such radical changes? i think it would be very nice if we could customize the user interface to this extent

I cannot understand how would such an interface improve the usability of the editor. [This](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fi.ytimg.com%2Fvi%2FpfVAfUEmNJs%2Fmaxresdefault.jpg&f=1&nofb=1) is "the editor", isn't it? Can you please elaborate?

> having a fun user interface, might actually make working with the engine far more pleasant, plus we can ignore convention and go truly crazy with the design and make everything easy and convenient to use, while making the editor look COOL AS HELL, it would force the designer to think outside the box, and we could create whatever our immagination is set loose to

Godot is not for designers only. Many of the roles in developing a game are unrelated to UI design. Moreover, the UI of the editor does not need to be cool in order to produce games which are cool. I'd rather see effort invested on features which allow to build cooler games/tools, rather than reworking the editor just because it is fancy. Of course, some cool and fancy tweaks can improve the usability/productivity at the same time. I'd be so glad to hear about those proposals. Thinking out of the box to make better use of the available space and to reduce the numbers of clicks/steps is what we need!

OTOH, convention and consistency save time, which is a very scarce resource. Covention can be ignored in assets, that's what they are for. But I would be really thankful if consistency accross platforms and versions was an important feature. IMHO, users should not learn what they already know again, unless it provides some added value.

> i would love to have some kind of easy "drag and drop" system, where you can just move things around,

Note that this is very different from theming. This is changing the template by reorganizing modules. I absolutely support this request... however, isn't moving things around possible already? Is doing it through drag & drop so important?

> i think allowing users to go wild with the user interface would be really nice, and i think the core idea is a good one

I'd say that users can already go wild with the user interface, through assets. Please can you be more specific about the wild things that you would like to do which you cannot currently?

> i think having an easy, visual editor, for the user interface, which allows you to go batshit crazy would be really great

Do you mean https://www.youtube.com/watch?v=Yonv0m5J3kQ or something similar to [Editor Theme Explorer](https://godotengine.org/asset-library/asset/557) which allows to modify and save the themes?

> plus, having such a system, would actually make creating user interfaces pleasant (for games, for the engine, everything)

Isn't it pleasant already?

> so the people who think, "THIS IS STUPID, THIS IS POINTLESS", no, you guys don't know what ur talking about, ur just ignoring a good idea, just because it's too different than what ur used to, but if u guys want to stick to the old ways, fine, that's your loss

Please, be respectful to other users. No one in this thread deemed the idea stupid or pointless and most of us know what we are talking about, at least as much as you do. We are not ignoring the idea; we reacted to it. The majority considered "not clear enough to be understood", and that's why we are explicitly "confused". That's feedback, which should let the OP know that he needs to either rethink or reword his proposal to better communicate his intent. It is disrespectful from you to dismiss that, just because apparently you don't like it.

OTOH, WinAmp alike customization is anything but too different or new. Precisely, it is an old idea from late 90s and early 00s. Please, have a look at the evolution of user interfaces (both on the desktop and on the web) since the late 90s until today. The vast majority of UI designers in the world might be awfully wrong, but please do argument why.

> and i'm sure i'll get the "BUT IT'S TOO MUCH HARD WORK" and "NO OTHER ENGINES DO THIS" excuses and, i just can't even be bothered to argue against those, if u can't see the value in this, or the potential benefits, and if i have to point out everything that's good about it, then i don't see the point in writing more

As you might understand, if you don't want to make an effort to expose and argue your points, there is little to no motivation for others taking you seriously. If it's too much work to properly propose something which you seem to understand clearly, imagine how much effort it takes to implement something that is undefined/undetermined.

> oh and, if everyone's arguments translate into, "IT'S TOO MUCH WORK", just, don't bother please, in my mind, it feels like u guys are just being too close minded for no reason, and ur too quick to shutdown an idea which takes some actual effort to implement

I wouldn't say it's too much effort. Because Godot is focused on generation of very custom interfaces, most of the resources are already there. However, I think it is not worth the effort.

Last, but not least, Godot is open source. You are entitled to pay someone to do any enhancements for you if don't know how.

**Comment by jasperbrooks79** [comment]
It was just an idea, I remember when a new WinAmp theme was there, I began using it again, or so . .

Also, over time if they were user-made, Godot could include some download links to nice ones, if they wanted, on official docs pages . .

It was just an idea, there is a new user-interface person, on the team, might be a fun way, to make Godot nicer, for instance there could be a fantasy theme, for RPG's, a sci-fi theme, for space games, however, it would have to be fun, and easy to use . .

Really liked WinAmp, skins made it a lot better, just fun . . But, it's more like a feature, if the other stuff is running well, priority is performance, and user-friendliness . . Anyway, it would be more, like a fun thing, the nicest ones might amuse the engine devs too, if it was user-made and, ran fast enough, to be decent performance . . . .

**Comment by jasperbrooks79** [comment]
Or, a theme that looked like classic, old games, Ultima games, for RPG makers, or Deux Ex theme, for sci - fi, remembering copyright stuff, anyway maybe the people working on user-interface could have this, as something to do, when major bugs, issues have been fixed, be really nice, with a custom looking engine, as one liked it, what-ever style . .

Tbc, this is a fun, not necessary thing, that shouldn't take away from far more important areas, I know . . . <3

**Comment by Calinou** [comment]
Closing due to lack of support (see reactions on OP).
